vllm/model_executor/layers/sampler.py

```diff
@@ -1262,6 +1264,10 @@ def _build_sampler_output(
         assert sample_logprobs is not None
         assert not isinstance(maybe_deferred_sample_results,
                               SampleResultArgsType)
+        assert len(sampling_metadata.seq_groups) \
+            == len(maybe_deferred_sample_results) \
+            == len(prompt_logprobs) \
+            == len(sample_logprobs)
```

Comment:
I'm not really familiar with the sampler code at all, but this assert at least does not trigger under any of our tests so far.

@comaniac @rickyyx Do y'all have pointers to any other places we should dig into where there might be an implicit assumption that the number of partial prefills is <= 1?

Reply:
I think the assertions currently would only happen if:
- All running sequences are chunked continuous prefills
- No decode sequences in the scheduler batch.

Reply:
Sweet, we will try to add some unit tests to cover that case and see if we can make it crash. Thanks!